src/Feature/Terminal/Feature_Terminal.re

```diff
@@ -406,3 +401,97 @@ let getLinesAndHighlights = (~colorTheme, ~terminalId) => {
      })
   |> Option.value(~default=([||], []));
 };
+
+// BUFFERRENDERER
+
+// TODO: Unify the model and renderer state. This shouldn't be needed.
+[@deriving show({with_path: false})]
+type rendererState = {
+  title: string,
+  id: int,
+  insertMode: bool,
+};
+
+let bufferRendererReducer = (state, action) => {
+  switch (action) {
+  | Service(ProcessTitleChanged({id, title, _})) when state.id == id => {
+      ...state,
+      id,
+      title,
+    }
+  | Command(NormalMode) => {...state, insertMode: false}
+  | Command(InsertMode) => {...state, insertMode: true}
+
+  | _ => state
+  };
+};
+
+// COMMANDS
+
+module Commands = {
+  open Feature_Commands.Schema;
+
+  module New = {
+    let horizontal =
+      define(
+        ~category="Terminal",
+        ~title="Open terminal in new horizontal split",
+        "terminal.new.horizontal",
+        Command(NewTerminal({cmd: None, splitDirection: Horizontal})),
+      );
+    let vertical =
+      define(
+        ~category="Terminal",
+        ~title="Open terminal in new vertical split",
+        "terminal.new.vertical",
+        Command(NewTerminal({cmd: None, splitDirection: Vertical})),
+      );
+    let current =
+      define(
+        ~category="Terminal",
+        ~title="Open terminal in current window",
+        "terminal.new.current",
+        Command(NewTerminal({cmd: None, splitDirection: Current})),
+      );
+  };
+
+  module Oni = {
+    let normalMode = define("oni.terminal.normalMode", Command(NormalMode));
+    let insertMode = define("oni.terminal.insertMode", Command(InsertMode));
+  };
+};
```

Comment:
An example of commands defined in and contributed from a feature project.

Reply:
Very cool 👍